core/primitives/res/runtime_configs/parameters.txt

```diff
@@ -0,0 +1,152 @@
+wasm_grow_mem_cost: 1
+wasm_host_function_base: 264768111
+wasm_contract_loading_base: 35445963
+wasm_contract_loading_bytes: 216750
+wasm_read_memory_base: 2609863200
+wasm_read_memory_byte: 3801333
+wasm_write_memory_base: 2803794861
+wasm_write_memory_byte: 2723772
+wasm_read_register_base: 2517165186
+wasm_read_register_byte: 98562
+wasm_write_register_base: 2865522486
+wasm_write_register_byte: 3801564
+wasm_utf8_decoding_base: 3111779061
+wasm_utf8_decoding_byte: 291580479
+wasm_utf16_decoding_base: 3543313050
+wasm_utf16_decoding_byte: 163577493
+wasm_sha256_base: 4540970250
+wasm_sha256_byte: 24117351
+wasm_keccak256_base: 5879491275
+wasm_keccak256_byte: 21471105
+wasm_keccak512_base: 5811388236
+wasm_keccak512_byte: 36649701
+wasm_ripemd160_base: 853675086
+wasm_ripemd160_block: 680107584
+wasm_ecrecover_base: 3365369625000
+wasm_log_base: 3543313050
+wasm_log_byte: 13198791
+wasm_storage_write_base: 64196736000
+wasm_storage_write_key_byte: 70482867
+wasm_storage_write_value_byte: 31018539
+wasm_storage_write_evicted_byte: 32117307
+wasm_storage_read_base: 56356845750
+wasm_storage_read_key_byte: 30952533
+wasm_storage_read_value_byte: 5611005
+wasm_storage_remove_base: 53473030500
+wasm_storage_remove_key_byte: 38220384
+wasm_storage_remove_ret_value_byte: 11531556
+wasm_storage_has_key_base: 54039896625
+wasm_storage_has_key_byte: 30790845
+wasm_storage_iter_create_prefix_base: 0
+wasm_storage_iter_create_prefix_byte: 0
+wasm_storage_iter_create_range_base: 0
+wasm_storage_iter_create_from_byte: 0
+wasm_storage_iter_create_to_byte: 0
+wasm_storage_iter_next_base: 0
+wasm_storage_iter_next_key_byte: 0
+wasm_storage_iter_next_value_byte: 0
+wasm_touching_trie_node: 16101955926
+wasm_read_cached_trie_node: 2280000000
+wasm_promise_and_base: 1465013400
+wasm_promise_and_per_promise: 5452176
+wasm_promise_return: 560152386
+wasm_validator_stake_base: 911834726400
+wasm_validator_total_stake_base: 911834726400
+wasm_alt_bn128_g1_multiexp_base: 713006929500
+wasm_alt_bn128_g1_multiexp_element: 3335092461
+wasm_alt_bn128_pairing_check_base: 9685508901000
+wasm_alt_bn128_pairing_check_element: 26575188546
+wasm_alt_bn128_g1_sum_base: 3175314375
+wasm_alt_bn128_g1_sum_element: 76218543
+
+# Smart contract limits
+max_gas_burnt: 200000000000000
+max_gas_burnt_view: 200000000000000
+max_stack_height: 16384
+stack_limiter_version: 0
+initial_memory_pages: 1024
+max_memory_pages: 2048
+registers_memory_limit: 1073741824
+max_register_size: 104857600
+max_number_registers: 100
+max_number_logs: 100
+max_total_log_length: 16384
+max_total_prepaid_gas: 300000000000000
+max_actions_per_receipt: 100
+max_number_bytes_method_names: 2000
+max_length_method_name: 256
+max_arguments_length: 4194304
+max_length_returned_data: 4194304
+max_contract_size: 4194304
+max_transaction_size: 4194304
+max_length_storage_key: 4194304
+max_length_storage_value: 4194304
+max_promises_per_function_call_action: 1024
+max_number_input_data_dependencies: 128
```

Comment:
Is there a design document around the configuration format? I feel like there may be alternatives that could possibly serve our use-cases slightly better. For example maybe a reverse object notation such as this

`wasm_config.limit_config.max_number_input_data_dependencies`?

would allow us to continue using serde infrastructure (though we'd need to implement a serde deserializer) or perhaps we could use something like `toml` or `json` and merge the files together.

In particular I had non-terrible experience with https://docs.rs/config/latest/config/ and TOML.

Reply:
Yeah, some bounded bikeshed would help here!

>would allow us to continue using serde infrastructure 

I wounder if we can go in the opposite direction and actually *remove* a bunch of serializes after this PR? Probably not, because, IIRC, we expose config-as-json over RPC API. If that's indeed the case, yeah, there's a strong reason to try to take advantage of serde. If we want to keep the current format (which I quite like, especialy if we add thousand separators), than we can potentially go `.txt -> serde_json::Value -> RuntimeConfig`

Reply:
@nagisa I have written a quick design-document in https://github.com/near/nearcore/issues/6710. I should have created an issue before submitting this PR, thanks for pointing it out! But I am also trying to keep things simple. Instead of putting too much time into finding the perfect solution, I would rather spend more time on writing tests that the chosen solution has no observable side-effects.

That said, I appreciate all inputs! As outlined in the other issue, keeping the file structured in any way was something I explicitly tried to avoid. The motivation that I even started to work on this was to avoid a lot of bloated code in PR https://github.com/near/nearcore/pull/6657 where @matklad suggested to first clean up the mess with parameters/profiled costs/runtime config before linking them to estimations. If you see a way to solve the problems I described in the other issue without fully-flattening, please let me know.

But I am open to use any sort of standardised format + serde. Although, I don't really see the value of it if we go the fully-flat route.